### PR TITLE
Bench2: Listener Cleanup, Minor Fixes, Manually Rescheduled Writes

### PR DESCRIPTION
**Pull request by simpsont-oci**
Cleanup Bench2's usage of DDS::Listener objects, including improving how we detach 'savvy' listeners from Bench2 builder library objects.

Also, somewhat unrelated, add the ability to prevent addition of random partition suffixes in the test_controller in order to more easily join a particular test run with an 'outside' process (e.g. one being run by valgrind / gdb) and added 'sm20' bench2 test to run_test.pl

Also, minor fixes to node_controller's signal handling and wait_for_discovery logic.

Also, add the ability to optionally configure "manually" rescheduled Writes via the WriteAction to avoid draining the Proactor's thread pool with locked pending writes.

Also, add the ability to request the POSIX AIO Proactor implementation to the worker config

### CODE REVIEW

File: performance-tests/bench_2/builder/Participant.cpp [hunk 1]
```diff
@@ -110,4 +111,20 @@ bool Participant::enable(bool throw_on_error) {
     publishers_->enable(throw_on_error);
 }
 
+void Participant::detach_listeners() {
```
Comment (mitza-oci) [inline performance-tests/bench_2/builder/Participant.cpp:114]:
```suggestion
void Participant::detach_listeners()
{
```
Not high priority, but it looks like some effort when in to style consistency elsewhere in this PR, so might as well apply here too.

File: performance-tests/bench_2/builder/PublisherManager.cpp [hunk 2]
```diff
@@ -2,19 +2,28 @@
 
 namespace Builder {
 
-PublisherManager::PublisherManager(const PublisherConfigSeq& configs, PublisherReportSeq& reports, DDS::DomainParticipant_var& participant, const std::shared_ptr<TopicManager>& topics, WriterMap& writer_map) {
+PublisherManager::PublisherManager(const PublisherConfigSeq& configs, PublisherReportSeq& reports, DDS::DomainParticipant_var& participant, const std::shared_ptr<TopicManager>& topics, WriterMap& writer_map)
```
Comment (sonndinh) [inline performance-tests/bench_2/builder/PublisherManager.cpp:5]:
```suggestion
PublisherManager::PublisherManager(const PublisherConfigSeq& configs, PublisherReportSeq& reports,
  DDS::DomainParticipant_var& participant, const std::shared_ptr<TopicManager>& topics, WriterMap& writer_map)
```

File: performance-tests/bench_2/builder/Topic.cpp [hunk 3]
```diff
@@ -129,4 +132,18 @@ bool Topic::enable(bool throw_on_error) {
   return result;
 }
 
+void Topic::detach_listener()
+{
+  if (listener_) {
+    TopicListener* savvy_listener_ = dynamic_cast<TopicListener*>(listener_.in());
+    if (savvy_listener_) {
+      savvy_listener_->unset_topic(*this);
+    }
```
Comment (sonndinh) [inline performance-tests/bench_2/builder/Topic.cpp:141]:
```suggestion
    TopicListener* savvy_listener = dynamic_cast<TopicListener*>(listener_.in());
    if (savvy_listener) {
      savvy_listener->unset_topic(*this);
    }
```

**Review by sonndinh — CHANGES_REQUESTED** [review]

File: performance-tests/bench_2/node_controller/main.cpp [hunk 4]
```diff
@@ -809,10 +809,11 @@ int run_cycle(
   }
 
   // Run Workers and Wait for Them to Finish
-  if (!worker_manager.run_workers(report_writer_impl)) {
+  if (!worker_manager->run_workers(report_writer_impl)) {
     std::cerr << "Running workers failed (likely because we received a SIGINT)" << std::endl;
     return 1;
   }
+  worker_manager.reset();
```
Comment (sonndinh) [inline performance-tests/bench_2/node_controller/main.cpp:816]: Shouldn't this be `worker_manager->reset()`?

Comment (simpsont-oci) [reply performance-tests/bench_2/node_controller/main.cpp:816]: No, it's resetting the shared pointer itself (destroying the WorkerManager in order to end the signal trapping). WorkerManager doesn't have a reset() method.

Comment (sonndinh) [reply performance-tests/bench_2/node_controller/main.cpp:816]: I see.

**Review by sonndinh — APPROVED** [review]

**Pull request merged**